src/editor/changesmanager.cpp

```diff
@@ -0,0 +1,114 @@
+#include "changemanager.h"
+
+ChangeManager::ChangeManager() = default;
+
+ChangeManager::ChangeManager(const std::string &fileState)
+{
+    currentFileState = fileState;
+}
+
+ChangeManager::~ChangeManager() = default;
+
+void ChangeManager::limitCheck()
+{
+    if(changesHistory.size() > CHANGES_HISTORY_MAX_SIZE)
+        changesHistory.pop_front();
+}
+
+void ChangeManager::removeHistory()//it's not necessary to hold history if we returned to the start and didn't press ctrl + Y
+{
+    if(!changesHistory.size())
+        return;
+    auto it_victim = changesHistory.end() - 1;// move from the end of list
+    while(it_victim != currentState_it)//delete all till pass iterator position(iterator isn't point to the end only if we pressed ctrl + Z)
+    {
+        it_victim--;
+        changesHistory.pop_back();
+    }
```

Comment:
this loop might be replaced with:
changesHistory.erase(std::next(currentState_it, 1), changesHistory.end());